tests/qa-activity-report/generate-qa-activity-report.go

```diff
@@ -0,0 +1,227 @@
+				changedTests := 0
+				for i, line := range diffLines {
+					if strings.Contains(line, "@@ Feature:") {
+						inScenarios, changed := checkChangedTests(diffLines, i)
+						if !inScenarios {
+							changedTests += changed
+						}
+					}
+				}
+			}
+		}
+
+		csvRows = append(csvRows, []string{"API Test", date, strconv.Itoa(addedTests), strconv.Itoa(changedTests), strconv.Itoa(deletedTests), commit})
+	}
+
+	// Ensure the directory exists
+	reportDir := "tests/qa-activity-report/reports"
+	if err := os.MkdirAll(reportDir, os.ModePerm); err != nil {
+		fmt.Println("Error creating directory:", err)
+		return
+	}
+
+	// Define the path for the CSV file
+	filePath := fmt.Sprintf("%s/QA_Activity_Report_%s.csv", reportDir, period)
+	if err := generateCSV(csvRows, filePath); err != nil {
+		fmt.Println("Error writing CSV report:", err)
+	} else {
+		fmt.Println("CSV report generated successfully. You can find it in", filePath)
+	}
+}
+
+func getTimeframe() (since string, until string, period string, err error) {
+	monthStr := os.Getenv("MONTH")
+	yearStr := os.Getenv("YEAR")
+	daysStr := os.Getenv("DAYS")
+
+	if monthStr != "" && yearStr != "" {
+		month, err := strconv.Atoi(monthStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid month: %w", err)
+		}
+		year, err := strconv.Atoi(yearStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid year: %w", err)
+		}
+		startDate := time.Date(year, time.Month(month), 1, 0, 0, 0, 0, time.UTC)
+		endDate := startDate.AddDate(0, 1, -1)
+		since = startDate.Format("2006-01-02")
+		until = endDate.Format("2006-01-02")
+		period = fmt.Sprintf("%02d_%04d", month, year)
+	} else if daysStr != "" {
+		days, err := strconv.Atoi(daysStr)
+		if err != nil {
+			return "", "", "", fmt.Errorf("invalid number of days: %w", err)
+		}
+		until = time.Now().Format("2006-01-02")
+		since = time.Now().AddDate(0, 0, -days).Format("2006-01-02")
+		period = fmt.Sprintf("Last_%d_days", days)
+	} else {
+		return "", "", "", fmt.Errorf("please provide either MONTH and YEAR or DAYS")
+	}
+
+	return since, until, period, nil
+}
+
+func getGitLog(since, until string) ([]string, error) {
+	cmd := exec.Command("git", "log", "--since="+since, "--until="+until, "--pretty=format:%H %ad", "--date=short", "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func getGitDiff(commit string) ([]string, error) {
+	cmd := exec.Command("git", "diff", commit+"~1", commit, "--", "tests/acceptance/features", ":(exclude)tests/acceptance/features/bootstrap/")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(out.String(), "\n"), nil
+}
+
+func countAddedTestsInExamples(diffLines []string, startIndex int) int {
+	var inExamples bool
+	inExamples = false
```

Comment:
```suggestion
	var inExamples bool
```

These two lines are equal, you can remove the second line.

Reply:
yes, I know. already tried to do it but sonarcloud wasn't happy

Reply:
I'll try again